dig_test.go

```diff
@@ -727,6 +876,21 @@ func TestInvokeFailures(t *testing.T) {
 		require.Contains(t, err.Error(), "dig.type2 isn't in the container")
 	})
 
+	t.Run("unmet named dependency", func(t *testing.T) {
```

Comment:
Could you add a similar test, but that has a type without a name provided?
Also reverse, there is a named type, but we ask for a default.